src/aurora.F

```diff
@@ -681,123 +551,155 @@ subroutine aurora_heat(lat)
-          alfa2(i+2) = alfa20
-          clat = acos(sin(abs(dlat_aur(i))))
-!         drizl(i+2) = exp(-((clat-crad(ihem)+
-!    |      abs(clat-crad(ihem)))/(2.*s10))**2)
-!         drizl(i+2) = exp(-((colat(i)-crad(ihem)+
-!    |      abs(colat(i)-crad(ihem)))/s10)**2)
-          flux(i+2) = max(efxg(i,lat)/(2.*alfa(i+2)*1.602e-9),1.e-20)
-!          flux2(i+2) = e20*(1.-re2*coslamda(i))*
-!     |      exp(-((clat-crad(ihem))/
-!!    |      exp(-((colat(i)-crad(ihem))/
-!     |      halfwidth(i))**2) / (2.*alfa2(i+2)*1.602E-9)
+            alfa(i,lat) = max(ekvg(im2,lat),1.)/2.
+            alfa2(i,lat) = alfa20
+            clat = acos(sin(abs(dlat_aur(i,lat))))
+            flux(i,lat) =
+     |        max(efxg(im2,lat)/(fac_p2e*alfa(i,lat)),1.e-20)
 ! New drizal formulation - G. Lu /08/17/2006
-          drizl(i+2) = exp(-((clat-rradp(ihem)+
-     |      abs(clat-rradp(ihem)))/(1*s10))**2)
-          flux2(i+2) = e20*(1.-re2*coslamda(i))*
-     |      exp(-((clat-rrad(ihem))/
-     |      halfwidth(i))**2) / (2.*alfa2(i+2)*1.602E-9)
-        endif
-
-      enddo ! i=1,nlon
-!
-! Periodic points:
-      do i=1,2
-        alfa  (i) = alfa (nlon+i)
-        flux  (i) = flux (nlon+i)
-	eflux (i) = eflux(nlon+i)
-        drizl (i) = drizl(nlon+i)
-        alfa2 (i) = alfa2(nlon+i)
-        alfa3 (i) = alfa3(nlon+i)
-        flux2 (i) = flux2(nlon+i)
-        flux3 (i) = flux3(nlon+i)
-	eflux2 (i) = eflux2(nlon+i)
-	eflux3 (i) = eflux3(nlon+i)
-        qteaur(i,lat)= qteaur(nlon+i,lat)
-        alfa  (nlonp2+i) = alfa (i+2)
-        flux  (nlonp2+i) = flux (i+2)
-	eflux (nlonp2+i) = eflux(i+2)
-        drizl (nlonp2+i) = drizl(i+2)
-        alfa2 (nlonp2+i) = alfa2(i+2)
-        alfa3 (nlonp2+i) = alfa3(i+2)
-	flux2 (nlonp2+i) = flux2(i+2)
-        flux3 (nlonp2+i) = flux3(i+2)
-	eflux2 (nlonp2+i) = eflux2(i+2)
-	eflux3 (nlonp2+i) = eflux3(i+2)
-        qteaur(nlonp2+i,lat)= qteaur(i+2,lat)
-      enddo
+            if (clat-rradp(ihem) > 0.) then
+              drizl(i,lat) = exp(-(2.*(clat-rradp(ihem))/s10)**2)
+            else
+              drizl(i,lat) = 1.
+            endif
+            flux2(i,lat) = e20*(1.-re2*coslamda)*
+     |        exp(-((clat-rrad(ihem))/halfwidth)**2)/
+     |        (fac_p2e*alfa2(i,lat))
+          endif
 !
 ! Replacing with M-I coupler values. Warning: alfa and flux can not be
 ! zero, otherwise the model will blow up. Set to small values here. This
 ! will not change simulations.
 !                                                        Wenbin  wang
 !
 #ifdef GAMERA
-      do i=1,nlonp4
-        alfa(i) = max(geng(i,lat),0.01)
-        flux(i) = max(gflx(i,lat),0.01)
-        eflux(i) = flux(i)*fac_p2e*alfa(i)
-!       write(6,*)alfa(i),flux(i),i,lat
-      enddo
+          alfa(i,lat) = max(geng(i,lat),0.01)
+          flux(i,lat) = max(gflx(i,lat),0.01)
+          eflux(i,lat) = flux(i,lat)*fac_p2e*alfa(i,lat)
 #endif
 
-      end subroutine aurora_heat
+! Sheng, 2017/07, Calculate SAPS ion drifts
```

Comment:
Should the Sheng SAPS ion drifts be turned off if GAMERA is turned on?

Reply:
Right, it should be off when GAMERA is on. Currently they are separate options.